packages/mdx/test/compile.js

```diff
@@ -548,6 +548,56 @@ test('compile', async () => {
   )
   console.log('\nnote: the preceding warning is expected!\n')
 
+  const developmentSourceNode = (
+    await run(
+      compileSync(
+        {value: '<div />', path: 'path/to/file.js'},
+        {development: true}
+      ).value
+    )
+  )({})
+  assert.equal(
+    // @ts-expect-error React attaches source information on this property,
+    // but it’s private and untyped.
+    developmentSourceNode._source,
+    {fileName: 'path/to/file.js', lineNumber: 1, columnNumber: 1},
+    'should expose source information in the automatic jsx dev runtime'
+  )
+
+  assert.equal(
```

Comment:
Maybe good to move this one next to the jsx tests (L826), in a new test for `automatic runtime` output?
Sometimes the output changes a bit and then its easier if all the output tests are next to eachother?